packages/idyll-components/src/text-input.js

```diff
@@ -1,25 +1,54 @@
-import React from 'react';
-const ReactDOM = require('react-dom');
-
+import React, { Fragment } from 'react';
 class TextInput extends React.PureComponent {
   constructor(props) {
     super(props);
     this.onChange = this.onChange.bind(this);
+    this.state = { error: { msg: '', isError: false } };
   }
 
   onChange(e) {
-    this.props.updateProps({ value: e.target.value });
+    const value = e.target.value || '';
+    const pattern = this.props.pattern;
+    const patternMismatchMessage =
+      this.props.patternMismatchMessage || `Input value doesn't match pattern`;
+
+    this.props.updateProps({ value: value });
+
+    if (pattern) {
+      try {
+        if (!value.match(pattern)) {
+          throw new Error(patternMismatchMessage);
+        }
+
+        eval(value); //check if there's JS syntax errors
```

Comment:
Can we drop this `eval()`? I can imagine cases where a user could reasonably be expected to input something that isn't valid JS source code. 

e.g. something like

```
[var name:"username" value:"" /]

Enter your name: [TextInput value:username /]

Your name is [Display value:username /]
```